io-reactor/src/reactor.rs

```diff
@@ -369,33 +383,78 @@ impl<H: Handler, P: Poll> Runtime<H, P> {
     fn handle_events(&mut self, time: Instant) -> bool {
         let mut awoken = false;
 
-        for (fd, io) in &mut self.poller {
+        for (fd, res) in &mut self.poller {
             if fd == self.waker.as_raw_fd() {
+                if let Err(err) = res {
+                    log::error!(target: "reactor", "Polling waker has failed: {err}");
+                    panic!("waker failure");
```

Comment:
Do we need to panic?

Reply:
What else can we do if the waker file descriptor is broken?